cutlass/group_mm.cu

```diff
@@ -0,0 +1,550 @@
+ */
+// clang-format on
+#include <cutlass_utils.h>
+#include <nvf_cutlass.h>
+
+#include <ATen/cuda/CUDAContext.h>
+#include <c10/cuda/CUDAGuard.h>
+#include <c10/cuda/CUDAStream.h>
+#include <cutlass/arch/arch.h>
+#include <torch/torch.h>
+
+#include <cassert>
+
+#include "cute/tensor.hpp"
+#include "cutlass/epilogue/collective/collective_builder.hpp"
+#include "cutlass/epilogue/collective/default_epilogue.hpp"
+#include "cutlass/epilogue/thread/linear_combination.h"
+#include "cutlass/gemm/collective/collective_builder.hpp"
+#include "cutlass/gemm/device/gemm_universal_adapter.h"
+#include "cutlass/gemm/dispatch_policy.hpp"
+#include "cutlass/gemm/group_array_problem_shape.hpp"
+#include "cutlass/gemm/kernel/gemm_universal.hpp"
+#include "cutlass/tensor_ref.h"
+#include "cutlass/util/command_line.h"
+#include "cutlass/util/distribution.h"
+#include "cutlass/util/host_tensor.h"
+#include "cutlass/util/packed_stride.hpp"
+#include "cutlass/util/reference/device/gemm.h"
+#include "cutlass/util/reference/device/tensor_compare.h"
+#include "cutlass/util/reference/host/gett.hpp"
+#include "cutlass/util/reference/host/tensor_compare.h"
+#include "cutlass/util/reference/host/tensor_fill.h"
+#include "cutlass/util/reference/host/tensor_norm.h"
+#include "cutlass/util/tensor_view_io.h"
+
+#include <exceptions.h>
+
+namespace nvfuser::cutlass_kernels {
+
+namespace {
+
+using namespace cute;
+
+#if defined(CUTLASS_ARCH_MMA_SM100_SUPPORTED)
+// Kernel configuration traits for different output data types
+// Defines tile shapes and cluster configurations.
+template <typename T, bool is_single_sm>
+struct KernelTraits;
+
+// Kernel traits for FP16 output
+template <>
+struct KernelTraits<cutlass::half_t, true> {
+  using MmaTileShape = Shape<_128, _256, Int<128 / sizeof(cutlass::half_t)>>;
+  using ClusterShape = Shape<_1, _1, _1>;
+  using KernelSchedule =
+      cutlass::gemm::KernelPtrArrayTmaWarpSpecialized1SmSm100;
+  using EpilogueSchedule = cutlass::epilogue::PtrArrayTmaWarpSpecialized1Sm;
+};
+
+// Kernel traits for BFloat16 output
+template <>
+struct KernelTraits<cutlass::bfloat16_t, true> {
+  using MmaTileShape =
+      Shape<_128, _256, Int<128 / sizeof(cutlass::bfloat16_t)>>;
+  using ClusterShape = Shape<_1, _1, _1>;
+  using KernelSchedule =
+      cutlass::gemm::KernelPtrArrayTmaWarpSpecialized1SmSm100;
+  using EpilogueSchedule = cutlass::epilogue::PtrArrayTmaWarpSpecialized1Sm;
+};
+
+// CUDA kernel to compute memory offsets and layout information for grouped GEMM
+// operations
+//
+// This kernel calculates the starting pointers and layout configurations for
+// each expert in a grouped matrix multiplication.
+//
+// Parameters:
+//   a_offsets: Output array of pointers to matrix A data for each expert
+//   b_offsets: Output array of pointers to matrix B data for each expert
+//   out_offsets: Output array of pointers to output matrix C data for each
+//   expert a_scales_offsets: Output array of pointers to A scaling factors for
```

Comment:
we don't have scales any more.
nitpick, the format could use some clean up here.